x-pack/legacy/plugins/siem/public/pages/case/components/all_cases/index.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import {
+  EuiBasicTable,
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiLoadingContent,
+  EuiTableSortingType,
+} from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import * as i18n from './translations';
+
+import { getCasesColumns } from './columns';
+import { SortFieldCase, Case, FilterOptions } from '../../../../containers/case/types';
+
+import { Direction } from '../../../../graphql/types';
+import { useGetCases } from '../../../../containers/case/use_get_cases';
+import { EuiBasicTableOnChange } from '../../../detection_engine/rules/types';
+import { Panel } from '../../../../components/panel';
+import { HeaderSection } from '../../../../components/header_section';
+import { CasesTableFilters } from './table_filters';
+
+import {
+  UtilityBar,
+  UtilityBarGroup,
+  UtilityBarSection,
+  UtilityBarText,
+} from '../../../../components/detection_engine/utility_bar';
+import { getCreateCaseUrl } from '../../../../components/link_to';
+
+export const AllCases = React.memo(() => {
+  const [
+    { data, isLoading, queryParams, filterOptions },
+    setQueryParams,
+    setFilters,
+  ] = useGetCases();
+
+  const tableOnChangeCallback = useCallback(
+    ({ page, sort }: EuiBasicTableOnChange) => {
+      let newQueryParams = {};
+      if (sort) {
+        let newSort;
+        switch (sort.field) {
+          case 'state':
+            newSort = SortFieldCase.state;
+            break;
+          case 'created_at':
+            newSort = SortFieldCase.createdAt;
+            break;
+          case 'updated_at':
+            newSort = SortFieldCase.updatedAt;
+            break;
+          default:
+            newSort = SortFieldCase.createdAt;
+        }
+        newQueryParams = {
+          ...newQueryParams,
+          sortField: newSort,
+          sortOrder: sort.direction as Direction,
+        };
+      }
+      if (page) {
+        newQueryParams = {
+          ...newQueryParams,
+          page: page.index + 1,
+          perPage: page.size,
+        };
+      }
+      setQueryParams(newQueryParams);
+    },
+    [setQueryParams, queryParams]
+  );
+
+  const onFilterChangedCallback = useCallback(
+    (newFilterOptions: Partial<FilterOptions>) => {
+      setFilters({ ...filterOptions, ...newFilterOptions });
+    },
+    [setFilters]
```

Comment:
filterOptions should be added as a dependency
```suggestion
    [filterOptions , setFilters]
```